### PR TITLE
Fix source handler bug where invalid ID was not caught soon enough

### PR DESCRIPTION
**Pull request by acrellin**
This PR fixes a bug where an invalid obj ID was not caught soon enough and was passed to `register_source_view` and `add_ps1_thumbnail` before erroring.

Closes https://github.com/skyportal/skyportal/issues/1372

### CODE REVIEW

**Review by kmshin1397 — APPROVED** [review]

**Comment by kmshin1397** [comment]
Looks good! Waiting on CI before merging

**Comment by kmshin1397** [comment]
@acrellin I see a bunch of sqlalchemy invalid request errors in the tests. They seem to be real seeing as I tried rerunning both CIs and got the same behavior. Seems like this change isn't as innocuous as it seems?

**Comment by acrellin** [comment]
@kmshin1397 these are also present on master; fixed in https://github.com/skyportal/skyportal/pull/1376

**Comment by kmshin1397** [comment]
@acrellin Ah okay great, I'll merge this then.

**Pull request merged**
